resolver/resolver.go

```diff
@@ -135,6 +157,12 @@ func (info *ResolverInfo) DescriptiveName() string {
 			info.Name,
 			info.ID(),
 		)
+	case info.IP == nil:
```

Comment:
Would recommend to check for `info.Domain != ""`, as this would prioritize the domain name if both Domain and IP are present.